docs/2-fork-accountability.md

```diff
@@ -0,0 +1,163 @@
+
+All the processes that are not detected are considered correct.
+
+## Monitor
+
+We introduce a trusted third-party verification entity which is responsible to run the accountability algorithm - we call it **monitor**.
+
+The monitor is responsible to coordinate and run the accountability algorithm as soon as a fork is detected. 
+The input of the algorithm are the message logs of the validators (i.e., all the information related to the exchanged messages by validators in the consensus instance). 
+The monitor will first request the message logs from each validator and then will run the accountability algorithm.
+
+We assume that all validators keep track of all the messages they sent and received during the execution of the Tendermint consensus algorithm. 
+The idea is that a correct process would be able to prove that it was not faulty by showing its activity (message logs) to the monitor.
+Indeed, the monitor is able to detect all the faulty processes that caused a fork (and not only) by analyzing the message logs from all the correct processes.
+
+## Fork scenarios
+
+In order to better understand the problem, we analyze the scenarios when the fork can happen and how the monitor would be able to handle the situation when analyzing the logs for each validator.
+
+### Fork in same round
+
+If a fork happens in a certain round r (two decisions are made in the same round r for value v and v'), we know that at least f+1 processes have sent a Prevote/Precommit message for both v and v'.
+Since at least one correct process will receive both messages, the monitor is able to infer all the "hidden" messages and eventually identify the faulty processes who caused the fork.  
+
+### Fork in different rounds
+
+If a fork happens in different rounds (value v is decided in round r1 and value v' is decided in round r2), we know that at least f+1 processes have sent a Precommit message for value v in round r and also sent a Prevote message for value v' in round r2 despite having locked value v when they sent a Precommit.
+
+Therefore, unless these processes have a justification (other 2f+1 Prevote messages for value v' in a round r'' s.t. r1 < r'' < r2), they are faulty. If they do have such a justification, then we now have to look for f+1 processes that sent Prevote messages for value v' despite having locked value v.
+
+As we can see, this is a recursive problem and we need to look back until we find f+1 processes that cannot justify the sending of a Prevote/Precommit message for another value despite having locked another one. 
+The method is simple: either a process is faulty because cannot justify the sending of a message or we go back recursively until we find processes that do not have a justification. It is clear that we will eventually find an iteration of the recursive problem where at least f+1 processes sent a Prevote message without a valid justification. 
```

Comment:
```suggestion
The method is simple: either a process is faulty because it cannot justify the sending of a message, or we go back recursively until we find processes that do not have a justification. It is clear that we will eventually find an iteration of the recursive problem where at least f+1 processes sent a Prevote message without a valid justification. 
```

Reply:
Fixed